leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/IntegrationTestHelper.java

```diff
@@ -246,6 +248,12 @@ public void deregisterClient() {
 
     public void dispose() {
         deregisterClient();
+        if (null != client) {
+            client.destroy(false);
+        }
+        if (null != server) {
+            server.destroy();
+        }
```

Comment:
We already call `stop()` in all tests. So we should replace it by `destroy()` or remove all the `stop()`call and `destroy` in dispose. Personally, I prefer the first solution as this make more sense to me to create in @before and destroy in @after.